core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1761,6 +1776,57 @@ private[spark] class BlockManager(
     blocksToRemove.size
   }
 
+  def decommissionBlockManager(): Unit = {
+    if (!blockManagerDecommissioning) {
+      logInfo("Starting block manager decommissioning process")
+      blockManagerDecommissioning = true
+      decommissionManager = Some(new BlockManagerDecommissionManager)
+      decommissionManager.foreach(_.start())
+    }
+  }
+
+  /**
+   * Tries to offload all cached RDD blocks from this BlockManager to peer BlockManagers
+   * Visible for testing
+   */
+  def offloadRddCacheBlocks(): Unit = {
+    val replicateBlocksInfo = master.getReplicateInfoForRDDBlocks(blockManagerId)
+
+    if (replicateBlocksInfo.nonEmpty) {
+      logInfo(s"Need to replicate ${replicateBlocksInfo.size} blocks " +
+        s"for block manager decommissioning")
+      // Refresh peer list once before starting replication
+      getPeers(true)
+    }
+
+    // Maximum number of storage replication failure which replicateBlock can handle
+    // before giving up for one block
```

Comment:
1. We had better have this message at the configuration declaration part.
2. We may need to revise the conf name to explicitly show this clause, `before giving up for one block`.

Reply:
@dongjoon-hyun Renamed to "spark.storage.decommission.maxReplicationFailuresPerBlock". Any suggestions on 2) ?